chatgpt-plugin/src/pages/api/handlers/solana-pay/sign/createTransferAsset.ts

```diff
@@ -0,0 +1,72 @@
+import {
+  SPL_ACCOUNT_COMPRESSION_PROGRAM_ID,
+  SPL_NOOP_PROGRAM_ID,
+} from "@solana/spl-account-compression";
+import { PublicKey, Transaction } from "@solana/web3.js";
+import { NextApiRequest } from "next";
+
+import { bs58 } from "@coral-xyz/anchor/dist/cjs/utils/bytes";
+import { createTransferInstruction } from "@metaplex-foundation/mpl-bubblegum";
+import { makeRespondToSolanaPayGet, makeRespondToSolanaPayPost } from ".";
+import configConstants, { CONNECTION } from "../../../constants";
+import { bufferToArray, getAsset, getAssetProof, getBubblegumAuthorityPDA } from "../utils/helpers";
+configConstants();
+
+async function createTransferAsset(req: NextApiRequest) {
+  const { destination, assetId } = req.query;
+  const { account: sender } = req.body;
+  if (!sender || !destination || !assetId) {
+    throw new Error("Missing required parameters");
+  }
+
+  const assetProof = await getAssetProof(assetId, CONNECTION.rpcEndpoint);
```

Comment:
please use https://github.com/helius-labs/helius-sdk

Reply:
didn't use the direct SDK because it uses a `wrappedConnection` which requires a signer and not just a public address. That's why had to refactor the methods to just work on the public keys.